src/Interpreters/Cache/FileCache.cpp

```diff
@@ -170,6 +187,35 @@ void FileCache::assertInitialized() const
 }
 
 void FileCache::initialize()
+{
+    // Prevent initialize() from running twice. This may be caused by two cache disks being created with the same path (see integration/test_filesystem_cache).
+    if (initialize_called.exchange(true))
+      return;
+
+    try
+    {
+      if (!fs::exists(getBasePath()))
```

Comment:
Please fix identation, it should be 4 spaces, not 2

Reply:
Done